### ISSUE TITLE
Minio bucket contents empty in Web UI

### ISSUE THREAD

**Issue by data-tangles** [minio/minio#19328]
## NOTE
If this case is urgent, please subscribe to [Subnet](https://min.io/pricing) so that our 24/7 support team may help you faster.

## Expected Behavior
Web UI shows the bucket data correctly

## Current Behavior
My Minio instance runs on a 2 node K3s cluster. I am using the latest container image version. I login to the Web UI successfully and can see the bucket present however it indicates that it is empty.

This bucket is configured to mount a local path on each of the K3s hosts. If I login to the hosts themselves the data is present. 

My Kubernetes manifest is as follows:

```
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: minio
  namespace: minio
  labels:
    app: minio
spec:
  replicas: 1
  selector:
    matchLabels:
      app: minio
  serviceName: minio
  template:
    metadata:
      labels:
        app: minio
    spec:
      containers:
      - name: minio
        image: minio/minio:latest
        args: ['server', '/data', '--console-address', ':9001']
        ports:
        - name: minio
          containerPort: 9000
        volumeMounts:
          - name: minio-data
            mountPath: /data
        env:
          - name: MINIO_SERVER_URL
            valueFrom:
              secretKeyRef:
                name: minio
                key: MINIO_SERVER_URL
          - name: MINIO_BROWSER_REDIRECT_URL
            valueFrom:
              secretKeyRef:
                name: minio
                key: MINIO_BROWSER_REDIRECT_URL
          - name: MINIO_ROOT_USER
            valueFrom:
              secretKeyRef:
                name: minio
                key: MINIO_ROOT_USER
          - name: MINIO_ROOT_PASSWORD
            valueFrom:
              secretKeyRef:
                name: minio
                key: MINIO_ROOT_PASSWORD
      volumes:
        - name: minio-data
          hostPath:
            path: /mnt/media/minio/data
```
Here is a screenshot showing the issue along with the 2 hosts which shows the data present

Web UI:

![Screenshot from 2024-03-25 08-59-07](https://github.com/minio/minio/assets/84679711/c43a6e83-8b41-419d-97c6-198eb05d695e)

K3s Host 1

![Screenshot from 2024-03-25 09-00-50](https://github.com/minio/minio/assets/84679711/6a71b970-4aec-47e8-b483-e86fd485112d)

K3s Host 2

![Screenshot from 2024-03-25 09-07-48](https://github.com/minio/minio/assets/84679711/c66df459-2107-4c50-a319-95c7012828f7)

## Steps to Reproduce (for bugs)

1. Utilise provided manifest
2. Setup a mount on underlying host
3. Create bucket with data 
4. Attempt to browse UI

## Context

I am utilising Minio for S3 type backups for Longhorn volumes for my other K3s containers. The consequence of this issue is that I can no longer list the backups within Longhorn itself to easily restore volumes

![Screenshot from 2024-03-25 09-02-54](https://github.com/minio/minio/assets/84679711/f1d84487-6080-4fc8-a071-9fa84a5a81d3)

## Your Environment

* Version used (`minio --version`):

minio version RELEASE.2024-03-21T23-13-43Z (commit-id=7fd76dbbb71eeba0dd1d7c16e7d96ec1a9deba52)

* Server setup and configuration:

Single Minio container that may be migrated between 2 hosts depending on load etc. Both hosts have the underlying mountpoint present so data would be intact no matter on which host the container resides. 

* Operating System and version (`uname -a`):
Ubuntu 22.04.4 LTS

**Comment by jiuker** [comment]
@data-tangles Can you first upload a file and see if it shows up in the list? If it can be listed, I personally feel that your storage may have lost the data.

**Comment by harshavardhana** [comment]
Existing data is not supported by MinIO. You need to reupload your data via MinIO for it to visible on the namespace.

**Comment by data-tangles** [comment]
> @data-tangles Can you first upload a file and see if it shows up in the list? If it can be listed, I personally feel that your storage may have lost the data.

Hi there. 

I did upload a file during my earlier tests and it did indeed appear. Minio itself actually shows that there are over 20000 files and 10 GB used in the Web UI

**Comment by data-tangles** [comment]
> Existing data is not supported by MinIO. You need to reupload your data via MinIO for it to visible on the namespace.

@harshavardhana 

Thanks for checking. 

I'm not sure I follow your statement. I can understand if it was brand new data that I'm trying to link into Minio but this is not the case. 

When I set this up last year I had created the "longhorn-backups" bucket at that time and then configured my Longhorn instance to point to this bucket via the s3 endpoint which was working fine. All the data within this bucket was created directly via Longhorn itself.

The last time I checked this I could see the data within the Minio browser. It's just when I checked on this recently that the data "dissapeared" but the underlying data is still present on the actual storage location on my file server.

As you will see below the Minio instance seems to "see" the existing data but just doesnt show it

![Screenshot from 2024-03-25 16-09-57](https://github.com/minio/minio/assets/84679711/d56126a4-5896-4ddb-a00c-6ee8dd27206b)
